rtcp-packet_test.go

```diff
@@ -0,0 +1,157 @@
+package rtp
+
+import (
+	"bytes"
+	"encoding/hex"
+	"testing"
+)
+
+
+func unreachable(t *testing.T) {
+  t.Fatalf("unreachable reached")
+}
+
+func assertEqual(t *testing.T, a interface{}, b interface{}) {
+	if a != b {
+    println("actual: ", a)
+    println("expected: ", b)
+		t.Fatalf("%s != %s", a, b)
+	}
+}
+
+func compareByteArrays(t *testing.T, actual []byte, expected []byte) {
+  if !bytes.Equal(actual, expected) {
+    t.Fatalf("Known-answer test failed: %x != %x", actual, expected)
+  }
+}
+
+
+func TestRTCPConstructGet(t *testing.T) {
+  p := NewRTCPacket(RTCPTypeSR/*pt*/, 1 /*length*/, 0xbcdc0094/*sender ssrc*/, []byte{1, 2, 3, 4} /*payload*/)
+
+  assertEqual(t, p.header.GetPT(), RTCPTypeSR)
+  assertEqual(t, p.header.GetLength(), uint16(1))
+  assertEqual(t, p.header.GetSenderSSRC(), uint32(0xbcdc0094))
+}
+
+func TestRTCPSet(t *testing.T) {
+  p := NewRTCPacket(RTCPTypeSR /*pt*/, 1 /*length*/, 0xbcdc0094/*sender ssrc*/, []byte{1, 2, 3, 4} /*payload*/)
+
+  p.header.SetPT(RTCPTypeRR)
+  p.header.SetLength(1)
+  p.header.SetSenderSSRC(0xbcdc1010)
+
+  assertEqual(t, p.header.GetPT(), RTCPTypeRR)
+  assertEqual(t, p.header.GetLength(), uint16(1))
+  assertEqual(t, p.header.GetSenderSSRC(), uint32(0xbcdc1010))
+}
+
+// Data from https://tools.ietf.org/html/rfc7714#section-17.2
+
+// salt: 51 75 69 64 20 70 72 6f 20 71 75 6f
```

Comment:
No need to have these values expressed twice.  Same comment for the other examples below.